source/features/useful-not-found-page.tsx

```diff
@@ -36,54 +39,53 @@ function parseCurrentURL(): string[] {
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
-	const parts = parseCurrentURL();
-	parts[2] = 'commits';
-	const url = '/' + parts.join('/');
-	if (await is404(location.origin + url)) {
+async function getLatestChangeToFile(): Promise<Record<string, any> | void> {
+	const {branch, filePath} = new GitHubURL(location.href);
+	if (!branch || !filePath) {
 		return;
 	}
 
-	bar.after(
-		<p className="container mt-4 text-center">
-			See also the {getType()}’s <a href={url}>commit history</a>.
-		</p>,
-	);
-}
-
-// If the resource exists in the default branch, link to it
-async function addDefaultBranchLink(bar: Element): Promise<void> {
-	const parts = getCleanPathname().split('/');
-	const branch = parts[3];
-	if (!branch) {
+	const {repository} = await api.v4(`
+		repository() {
+			object(expression: "${branch}") {
+				... on Commit {
+					history(first: 1, path: "${filePath}") {
+						nodes {
+							oid
+						}
+					}
+				}
+			}
+		}
+	`);
+	const commit = repository.object?.history.nodes[0];
+	if (!commit) {
 		return;
 	}
 
-	const defaultBranch = await getDefaultBranch();
-	if (branch === defaultBranch) {
-		return;
+	const commitInfo = await getCommitInfo(commit.oid as string);
+	for (const fileInfo of commitInfo.files) {
+		if ([fileInfo.filename, fileInfo.previous_filename].includes(filePath)) {
+			return {fileInfo, commitInfo};
+		}
 	}
```

Comment:
Can you move this into `getCommitInfo` itself? Both usages look the files array to find one of the files. If it's not found, it should return `undefined`

```ts
function getRenamedFile(commitHash, filePath): Record<string, any> | undefined
```

Also ideally we should return a more specific object than `Record<string, any>`

Reply:
Moved that part into the `getRenamedFile` function, renamed that function to `getChangesToFileInCommit(sha, filePath)` and added a `CommitInfo` type in [my fork](https://github.com/gamemaker1/refined-github/commit/68654239c21cf3c583b5e7ae006323c6790fd383), should I open a new PR for this?

Reply:
Yes